open-craft/opencraft
# [SE-5023] Integrate with Grove

## Description

This pull requests integrates the new [Grove](https://grove.opencraft.com/) based deployments into Ocim (console backend). The integration involves triggering GitLab pipelines of Grove to create new instances and receiving webhooks from the pipelines, informing about build status.

### Dependencies

N/A

### Screenshots

N/A

## Sandbox

TBD -- Here comes an instance deployed by Grove

## Testing instructions

1. Deploy the changes to stage and ensure all migrations are applied
2. Ensure the Grove `.env` settings are set
3. Register a new account and confirm email address
4. Check the pipeline is started in GitLab
5. _coffee break -- the deployment takes ~25 minutes_
6. Check the deployment succeed
7. Check the deployment status on the Client Console
8. Start a new deployment from the client console

## Deadline

ASAP

## TODO

- [x] Resolve `No serializer enabled for that Instance type` error
- [x] Resolve `InstanceReference matching query does not exist.`
- [x] Slugify instance name when sending the pipeline triggering
- [x] Rebase onto master
- [x] Add pipeline object to Django admin

## Other information

This PR is created by multiple members, therefore none of the involved developers can approve this PR to satisfy the GitHub checkes; therefore we are going to use our "administrator privileges" to merge the PR.
